### PR TITLE
sync launch settings

### PR DESCRIPTION
**Pull request by philanderson888**

### CODE REVIEW

**Comment by philanderson888** [comment]
OK @TwitchBronBron looks like it's passing all tests so over to you ... thanks.

File: package.json [hunk 1]
```diff
@@ -1592,6 +1595,306 @@
                         "type": "string",
                         "description": "A path to a file where all brightscript console output will be written. If null or empty, file logging will be disabled.",
                         "scope": "resource"
+                    },
+                    "brightscript.debug.host": {
```
Comment (TwitchBronBron) [inline package.json:1599]: All of these need to have `"scope": "resource"` added to them. This should happen in the sync-launch-settings.ts script.

File: package.json [hunk 2]
```diff
@@ -600,11 +600,14 @@
                             },
                             "logfilePath": {
                                 "type": "string",
-                                "description": "A path to a file where all brightscript console output will be written. If null or empty, file logging will be disabled."
+                                "description": "A path to a file where all brightscript console output will be written. If null or empty, file logging will be disabled.",
+                                "scope": "resource"
```
Comment (TwitchBronBron) [inline package.json:604]: `"scope": "resource"` doesn't belong in debug configuration settings. Remove this entry.

File: package.json [hunk 3]
```diff
@@ -600,11 +600,14 @@
...
                             },
                             "enableDebugProtocol": {
                                 "type": "boolean",
-                                "description": "If true, the debugger will use the new BrightScript debug protocol, and will disable the telnet debugger. See this link for more details: https://developer.roku.com/en-ca/docs/developer-program/debugging/socket-based-debugger.md"
+                                "default": false,
+                                "description": "If true, the debugger will use the new BrightScript debug protocol, and will disable the telnet debugger. See this link for more details: https://developer.roku.com/en-ca/docs/developer-program/debugging/socket-based-debugger.md",
+                                "scope": "resource"
```
Comment (TwitchBronBron) [inline package.json:610]: `"scope": "resource"` doesn't belong in debug configuration settings. Remove this entry.

File: package.json (hunk 1 above)
Comment (philanderson888) [reply package.json:1599]: Will do ...

File: package.json (hunk 2 above)
Comment (philanderson888) [reply package.json:604]: done, thanks

File: package.json (hunk 3 above)
Comment (philanderson888) [reply package.json:610]: done, thanks

File: package.json (hunk 1 above)
Comment (philanderson888) [reply package.json:1597]: changes made @TwitchBronBron thanks

File: package.json (hunk 2 above)
Comment (philanderson888) [reply package.json]: removed thanks

File: package.json (hunk 3 above)
Comment (philanderson888) [reply package.json]: removed thanks

**Comment by philanderson888** [comment]
All amendments made - please check again @TwitchBronBron thanks

File: package.json (hunk 1 above)
Comment (TwitchBronBron) [reply package.json:1597]: @philanderson888 The `"scope": "resource"` seems to still be missing in these.

Comment (philanderson888) [reply package.json:1597]: Ah yes... That will be because I did not re-run the script yet!!!

Comment (philanderson888) [reply package.json:1598]: OK @TwitchBronBron I have now re-run the script.  All of the 'brightscript.debug.' entries now contain "scope":"resource" in them, populated from the script.

**Review by TwitchBronBron — APPROVED** [review]

**Comment by TwitchBronBron** [comment]
Looks great! Thanks for this.

**Comment by philanderson888** [comment]
OK!  Happy to be of help!

**Pull request merged**